x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/create_datasource_page/index.tsx

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState } from 'react';
+import {
+  useRouteMatch,
+  HashRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+  useHistory,
+} from 'react-router-dom';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiButtonEmpty } from '@elastic/eui';
+import { AGENT_CONFIG_DETAILS_PATH } from '../../../constants';
+import { AgentConfig, PackageInfo, NewDatasource } from '../../../types';
+import { useLink, sendCreateDatasource } from '../../../hooks';
+import { useLinks as useEPMLinks } from '../../epm/hooks';
+import { CreateDatasourcePageLayout } from './components';
+import { CreateDatasourceFrom, CreateDatasourceStep } from './types';
+import { CREATE_DATASOURCE_STEP_PATHS } from './constants';
+import { StepSelectPackage } from './step_select_package';
+import { StepSelectConfig } from './step_select_config';
+import { StepConfigureDatasource } from './step_configure_datasource';
+import { StepReviewDatasource } from './step_review';
+
+export const CreateDatasourcePage: React.FunctionComponent = () => {
+  const {
+    params: { configId, pkgkey },
+    path: matchPath,
+    url: basePath,
+  } = useRouteMatch();
+  const history = useHistory();
+  const from: CreateDatasourceFrom = configId ? 'config' : 'package';
+  const [maxStep, setMaxStep] = useState<CreateDatasourceStep | ''>('');
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  // Agent config and package info states
+  const [agentConfig, setAgentConfig] = useState<AgentConfig>();
+  const [packageInfo, setPackageInfo] = useState<PackageInfo>();
+
+  // New datasource state
+  const [datasource, setDatasource] = useState<NewDatasource>({
+    name: '',
+    description: '',
+    config_id: '',
+    enabled: true,
+    output_id: '', // TODO: Blank for now as we only support default output
+    inputs: [],
+  });
+
+  // Update package info method
+  const updatePackageInfo = (updatedPackageInfo: PackageInfo | undefined) => {
+    if (updatedPackageInfo) {
+      setPackageInfo(updatedPackageInfo);
+    } else {
+      setPackageInfo(undefined);
+      setMaxStep('');
+    }
+
+    // eslint-disable-next-line no-console
+    console.debug('Package info updated', updatedPackageInfo);
```

Comment:
remove prior to merge? (same below)

Reply:
I'd like to keep these debug lines here for a little while to make it easy to troubleshoot issues as more packages are built out. We should definitely remove them before release